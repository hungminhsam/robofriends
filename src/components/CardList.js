import React from "react";
import Card from "./Card";

const CardList = ({ data }) => {
  return data.map(({ id, name, email }) => (
    <Card key={id} id={id} name={name} email={email} />
  ));
};

export default CardList;
