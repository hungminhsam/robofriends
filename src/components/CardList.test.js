import { shallow } from "enzyme";
import React from "react";
import { CardList } from "./CardList";

describe("<CardList />", () => {
  it("expect to render CardList", () => {
    const mockProps = {
      searchText: "",
      robots: {
        isPending: false,
        error: null,
        data: [],
      },
    };
    expect(shallow(<CardList {...mockProps} />).debug()).toMatchSnapshot();
  });

  it("expect to render isPending", () => {
    const mockProps = {
      searchText: "",
      robots: {
        isPending: true,
        error: null,
        data: [],
      },
    };
    expect(shallow(<CardList {...mockProps} />).text()).toEqual("Loading...");
  });

  it("expect to rander error message", () => {
    const mockProps = {
      searchText: "",
      robots: {
        isPending: false,
        error: { message: "error message" },
        data: [],
      },
    };
    expect(shallow(<CardList {...mockProps} />).text()).toEqual(
      "error message"
    );
  });

  it("expect to render all robots correctly", () => {
    const mockProps = {
      searchText: "",
      robots: {
        isPending: false,
        error: null,
        data: [
          {
            id: 1,
            name: "Jane Doe",
            email: "jane.doe@gmail.com",
          },
          {
            id: 2,
            name: "Jone Doe",
            email: "jone.doe@gmail.com",
          },
          {
            id: 3,
            name: "Joe Bloke",
            email: "joe.block@gmail.com",
          },
        ],
      },
    };
    expect(shallow(<CardList {...mockProps} />).find("Card").length).toEqual(3);
  });

  it("expect to filter correctly", () => {
    const mockProps = {
      searchText: "Jane",
      robots: {
        isPending: false,
        error: null,
        data: [
          {
            id: 1,
            name: "Jane Doe",
            email: "jane.doe@gmail.com",
          },
          {
            id: 2,
            name: "Jone Doe",
            email: "jone.doe@gmail.com",
          },
          {
            id: 3,
            name: "Joe Bloke",
            email: "joe.block@gmail.com",
          },
        ],
      },
    };
    expect(shallow(<CardList {...mockProps} />).find("Card").length).toEqual(1);
  });
});
