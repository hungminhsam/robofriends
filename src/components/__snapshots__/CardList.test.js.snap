// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> expect to render CardList 1`] = `"<ErrorBoundary />"`;
